tests/api/api/test_utils.py

```diff
@@ -295,6 +306,410 @@ def test_generate_function_and_task_from_submit_run_body_keep_credentials(
     assert parsed_function_object.metadata.credentials.access_key == access_key
 
 
+def test_ensure_function_has_auth_set(
+    db: Session, client: TestClient, k8s_secrets_mock: tests.api.conftest.K8sSecretsMock
```

Comment:
This function and the followings are very long, replacing the comments with `log.info()` would give better context when reviewing the tests logs.